contracts/JBETHPaymentTerminal.sol

```diff
@@ -302,24 +302,48 @@ contract JBETHPaymentTerminal is
     // and receive any extra distributable funds not allocated to payout splits.
     address payable _projectOwner = payable(projects.ownerOf(_projectId));
 
-    // Take a fee from the _distributedAmount, if needed.
-    // The project's owner will be the beneficiary of the resulting minted tokens from platform project.
-    // The platform project's ID is 1.
-    uint256 _feeAmount = fee == 0 || _projectId == 1
-      ? 0
-      : _takeFeeFrom(_projectId, _fundingCycle, _distributedAmount, _projectOwner);
+    // Define variables that will be needed outside scoped section below.
+    uint256 _feeAmount;
+    uint256 _leftoverDistributionAmount;
 
-    // Payout to splits and get a reference to the leftover transfer amount after all mods have been paid.
-    // The net transfer amount is the withdrawn amount minus the fee.
-    uint256 _leftoverDistributionAmount = _distributeToPayoutSplitsOf(
-      _projectId,
-      _fundingCycle,
-      _distributedAmount - _feeAmount
-    );
+    // Scoped section prevents stack too deep. _feeDiscount and _feeEligibleDistributionAmount only used within scope.
+    {
+      // Get the amount of discount that should be applied to any fees take.
+      uint256 _feeDiscount = _getFeeDiscount(_projectId);
 
-    // Transfer any remaining balance to the project owner.
-    if (_leftoverDistributionAmount > 0)
-      Address.sendValue(_projectOwner, _leftoverDistributionAmount);
+      uint256 _feeEligibleDistributionAmount;
+
+      // Payout to splits and get a reference to the leftover transfer amount after all mods have been paid.
+      // Also get a reference to the amount that was distributed that is eligible to have fees taken.
+      // The net transfer amount is the withdrawn amount minus the fee.
+      (_leftoverDistributionAmount, _feeEligibleDistributionAmount) = _distributeToPayoutSplitsOf(
+        _projectId,
+        _fundingCycle,
+        _distributedAmount,
+        _feeDiscount
+      );
+
+      // Leftover distribution amount is also eligible for a fee since the funds are going out of the ecosystem to _beneficiary.
+      _feeEligibleDistributionAmount += _leftoverDistributionAmount;
+
+      // Take the fee.
+      _feeAmount = fee == 0 || _feeEligibleDistributionAmount == 0 || _projectId == 1
```

Comment:
Shouldn't we reorg this for "probabilistic gas saving"?
I'd say the order: _projectId == 1 || _feeEligibleDistributionAmount == 0 || fee==0

Reply:
dope